build Docker image and publish to GitHub registry

marzocchi opened the pull request:
This adds a workflow to build `erd`'s Docker image and publish it to GitHub's registry and resolves #104.

To Do

- [ ] choose appropriate tags
- [ ] choose when to trigger (eg. push to master, release, etc)

marzocchi commented:
hi @mmzx, @BurntSushi!

I thought this might help making `erd` more approachable, as building locally can take quite some time and resources. It's not complete (see To Do), but let me know if you are interested in merging this PR and your preference for tags and when to trigger it.

To note:

- it's basically just the [workflow from the docs](https://docs.github.com/en/actions/guides/publishing-docker-images#publishing-images-to-github-packages)
- uses GitHub's registry and not Docker's as it requires no setup by you (eg. no need to setup secrets as it uses [the built-in one](https://docs.github.com/en/actions/reference/authentication-in-a-workflow#about-the-github_token-secret)) 
- you can [see here how it would look like](https://github.com/marzocchi/erd/pkgs/container/erd)


And this is an usage example:

```
docker run -i ghcr.io/marzocchi/erd:latest <<EOF >| out.pdf
[Person]
*name
+birth_place_id
EOF
```

mmzx commented:
I think the idea is nice. The choice of base image is important, to have the necessary ghc version. Let me follow up...

marzocchi commented:
sure! well anyway the workflow builds the `Dockerfile` from the root of the repo, so `haskell:8`

fbartho commented:
@marzocchi / @mmzx — did this work ever reach completion? I’m interested in pulling this tool or something like it in, for generating DB Schema Images in our Github PRs

marzocchi commented:
hi @fbartho, there was no further development but I'm ready to wrap it up as soon as @mmzx's follows up on the choice of base image

abbluiz commented:
I'm curious whether the image will support arm architecture

mmzx commented:
Sorry for slacking off here! - somehow missed it.
I've made some experiment today using the `haskell:8` base image, that could be used for the purpose of the PR. Anyone, who would be still interested doing it?

marzocchi commented:
hi @mmzx, I updated the workflow's dependencies and it now builds when a tag matching `vX.X.X` is pushed (also works when making a release on GitHub as long as the expected tag is created). The image will be tagged with `latest` and the tag for which it was built (see `v0.0.4` [here](https://github.com/marzocchi/erd/pkgs/container/erd) for an example).

I also updated the docker example in the README.

Note that the Dockerfile used by the workflow is the one already existing in the root of the repo, which already uses `haskell:8`.

mmzx commented:
Brilliant, thank you very much!

FelixVanOost commented:
@mmzx Are there any plans to release a new tag in the forseeable future so that the workflow merged in this PR gets triggered?

If so, I would like to submit a PR to additionally build and release `arm64` variants of the Docker images based on the work done here.

mmzx commented:
> 

@FelixVanOost , go ahead with your PR. I will trigger the release build then!
